pkg/imagehandler/imagefile.go

```diff
@@ -26,12 +28,25 @@ type imageFile struct {
 	size              int64
 	ignitionContent   []byte
 	rhcosStreamReader io.ReadSeeker
+	initramfs         bool
 }
 
 // file interface implementation
 
 var _ fs.File = &imageFile{}
 
+func (f *imageFile) Init(inputFile baseFile) error {
+	if f.rhcosStreamReader == nil {
+		var err error
+		ignition := &isoeditor.IgnitionContent{Config: f.ignitionContent}
+		f.rhcosStreamReader, err = inputFile.InsertIgnition(ignition)
```

Comment:
This here looks a bit confusing, something like the following?
```suggestion
		f.finalImage, err = inputFile.InsertIgnition(ignition)
```